pkg/query/flamegraph_arrow.go

```diff
@@ -1207,6 +1247,262 @@ func (fb *flamegraphBuilder) addRowValues(r *profile.RecordReader, row, sampleRo
 	fb.builderDiff.Add(row, r.Diff.Value(sampleRow))
 }
 
+func (fb *flamegraphBuilder) trim(ctx context.Context, tracer trace.Tracer, threshold float32) error {
+	_, span := tracer.Start(ctx, "trim")
+	defer span.End()
+
+	releasers := make([]releasable, 0, 3)
```

Comment:
We should be able to calculate the correct amount upfront now. 
```suggestion
	releasers := make([]releasable, 0, 10 + 2 * len(fb.labels))
```
We can follow up with that in another PR. Shouldn't be bad.

Reply:
Done